### ISSUE TITLE
PyuvmFormatter is not multi-handler friendly

### ISSUE THREAD

**Issue by PabloSML** [pyuvm/pyuvm#239]
The `format` method in PyuvmFormatter modifies the record (logging.LogRecord) object instead of creating a copy ([reference](https://github.com/pyuvm/pyuvm/blob/a7e5a13262c062be858eb43a26ffd38e41c7b087/pyuvm/s06_reporting_classes.py#L56)).

This means when using `add_logging_handler` with a handler without a custom formatter, the second formatter to get the record for processing will receive a message with the `[{self.full_name}]` already added in the beginning, duplicating the information.

A `deepcopy` of the record should be done in the beginning of the `format` method to ensure each handler does its own formatting independently, or a `msg_temp` variable should be used to restore original message after use.

**Comment by LeoGoTi** [comment]
Agreed. I had a similar issue as reported [here](https://github.com/pyuvm/pyuvm/issues/214#issuecomment-2196578228).

My problem was due to multiple tests declared in the same python file resulting in tests run in sequence in one `simv` process lifetime, in addition to multiple logger registered somehow for the same `get_full_name()` path.

I believe that some amendments is necessary here. One more option may be checking for `self.full_name` substring existence in `record.msg`, and avoiding attaching it twice if detected.

**Comment by raysalemi** [comment]
Please make a simple example of the problem so that we can fashion a solution. Thanks!

**Comment by PabloSML** [comment]
### Example problem code
```python
from pyuvm import *
import logging

file_handler = logging.FileHandler('cocotb_output.log')
uvm_root().add_logging_handler(file_handler)
uvm_root().logger.info("This message will have duplicated name tags")
```
The log file will have a message similar to the following:
```
0.00ns INFO   [uvm_test_top]: [uvm_test_top]: This message will have duplicated name tags
```

### Possible solution
```python
class PyuvmFormatter(SimColourLogFormatter):
    def __init__(self, full_name):
        self.full_name = full_name
        super().__init__()

    def format(self, record):
        msg_temp = record.msg   # Save original message here
        new_msg = f"[{self.full_name}]: " + record.msg
        record.msg = new_msg
        name_temp = record.name
        record.name = f"{record.pathname}({record.lineno})"
        if want_color_output():
            formatted_msg = super().format(record)
        else:
            formatted_msg = SimLogFormatter.format(self, record)
        record.msg = msg_temp    # Restore original message to avoid name duplication
        record.name = name_temp
        return formatted_msg
```

If you need a runnable example for testing instead of this simple example, I could write one at a later date. I could also fork and PR the solution at your request.

Thanks!

**Comment by raysalemi** [comment]
I have merged the pull request since it passes all the tests.  Does the latest version of pyuvm solve the problem?
